ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx

```diff
@@ -1,29 +1,80 @@
+import React, { useMemo, useState } from 'react';
 import {
     Bullseye,
     Button,
     Divider,
+    Flex,
+    FlexItem,
     PageSection,
     Pagination,
-    SearchInput,
+    Select,
+    SelectOption,
     Spinner,
     Text,
     Title,
     Toolbar,
     ToolbarContent,
+    ToolbarGroup,
     ToolbarItem,
+    debounce,
 } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { useQuery } from '@apollo/client';
+import cloneDeep from 'lodash/cloneDeep';
 
 import PageTitle from 'Components/PageTitle';
 import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate/EmptyStateTemplate';
-import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import SearchFilterChips from 'Components/PatternFly/SearchFilterChips';
+import { searchCategories } from 'constants/entityTypes';
 import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import { searchValueAsArray } from 'utils/searchUtils';
 import useURLPagination from 'hooks/useURLPagination';
 import useURLSort from 'hooks/useURLSort';
 import useURLSearch from 'hooks/useURLSearch';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import SEARCH_AUTOCOMPLETE_QUERY from 'queries/searchAutocomplete';
+import { SearchFilter } from 'types/search';
 import { useDeploymentListeningEndpoints } from './hooks/useDeploymentListeningEndpoints';
 import ListeningEndpointsTable from './ListeningEndpointsTable';
 
+import './ListeningEndpointsPage.css';
+
+/**
+ * Return request query string for autocomplete searches. Results will be scoped based
+ * on applied filters that are of a higher scope than the autocomplete category.
+ * e.g. If the user is filtering by Namespace, the autocomplete results will be scoped
+ *      to the the selected Cluster, if it exists. Filters of the same scope or lower
+ *      will not be included in the query string.
+ * @param searchFilter The current search filter
+ * @param category The category of the autocomplete search
+ * @param value The value of the autocomplete search
+ * @returns The request query string
+ */
+export function getRequestQueryStringForAutocomplete(
+    searchFilter: SearchFilter,
+    category: string,
+    value: string
+): string {
+    const filter = cloneDeep(searchFilter);
+
+    // Do not include lower scoped filters in the query string
+    if (category === 'Cluster' || category === 'Namespace') {
+        delete filter.Deployment;
+    }
+    if (category === 'Cluster') {
+        delete filter.Namespace;
+    }
+    // Do not include any existing filters for the autocomplete category
+    // e.g. If the user is filtering by Namespace, do not include the Namespace
+    //      filter in the query string
+    delete filter[category];
+
+    return Object.entries(filter)
+        .map(([key, val]) => `${key}:${Array.isArray(val) ? val.join(',') : val ?? ''}`)
+        .concat(`${category}:${value}`)
+        .join('+');
```

Comment:
How does this snippet (that seemed familiar) compare with:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/searchUtils.ts#L134-L139

Reply:
This snippet was derived from that search util. I had thought about generalizing the two, but with the `filter` call in one, and the `concat` in the other, plus the surrounding logic, it didn't feel worth the effort to abstract for the 3 lines shared in common.

Reply:
Yes, I agree.